test/buyer.test.js

```diff
@@ -61,19 +64,77 @@ describe("testing buyer signup",() =>{
   describe("tesing signin email and password",() =>{
     let cookie;
     test('user signin',async () =>{
-      const res = await request(app).post("/api/user/login").send({
+      const res = await request.post("/api/user/login").send({
         email: "ngarukiyimanasostene@gmail.com",
         password: "1234567@password"
       });
       expect(res.statusCode).toBe(200);
       expect(res.body.status).toBe('success');
+      token = res.body.token;
 
       cookie = res.headers['set-cookie'][0];
 
   });
+  describe('/addToWishlist/:productId endpoint', () => {
+
+    it('should return 404 if product is not found', async () => {
+  
+      const res = await request.post('/api/product/addToWishlist/100').set('token', `Bearer ${token}`);
+      expect(res.statusCode).toBe(404);
+      expect(res.body).toEqual({ message: 'Product not found' });
+    });
+  
+    it('should create a new wishlist if one does not exist', async () => {
+  
+      const productId = 1;
+  
+      const res = await request.post(`/api/product/addToWishlist/${productId}`).set('token', `Bearer ${token}`);
+      expect(res.statusCode).toBe(200);
+    
+});
+  
+    it('should add a product to an existing wishlist', async () => {
+  
+       const prodId = 2;
+  
+      const res = await request.post(`/api/product/addToWishlist/${prodId}`).set('token',  `Bearer ${token}`);
+      expect(res.statusCode).toBe(200);
+    
+  
+    });
+  
+    it('should return 400 if product is already in wishlist', async () => {
+      const proId = 2;
+  
+      const res = await request.post(`/api/product/addToWishlist/${proId}`)
+        .set('token', `Bearer ${token}`)
+      expect(res.statusCode).toBe(400);
+  
+      expect(res.body).toEqual({ message: 'Product already in wishlist' });
+  
+    });
+  });
+
+
+  
+  describe('/retrieveWishlistItems endpoint', () => {
+  
+    it('should retrieve wishlist items', async () => {
+  
+      const res = await request.get('/api/product/retrieveWishlistItems').set('token',  `Bearer ${token}`);
+      expect(res.statusCode).toBe(200);
+      expect(res.body).toEqual({ wishlist: expect.any(Array) });
+    });
+  
+    it('should return array of all products', async () => {
+        const res = await request.get('/api/product/retrieveWishlistItems').set('token',  `Bearer ${token}`);
+        expect(res.statusCode).toBe(200);
+       
+    });})
+      
```

Comment:
you could use the `beforeAll ` hook in Jest to obtain a token before running all tests:
example:
`
beforeAll(async () => {
  const res = await request.post("/api/user/login").send({
    email: "ngarukiyimanasostene@gmail.com",
    password: "1234567@password"
  });
  token = res.body.token;
});
`
the `beforeAll` hook runs once before all tests, and sends a login request to the server to obtain a token. The token is then assigned to the token variable, which is used in subsequent requests using the `set `method on the `request `object to add the token to the request headers.